lib/Cron/Workers/SimpleWorker.php

```diff
@@ -42,7 +43,7 @@ function process() {
     $scheduled_tasks = self::getScheduledTasks();
     $running_tasks = self::getRunningTasks();
 
-    if (!$scheduled_tasks && !$running_tasks) {
+    if (!$scheduled_tasks && !$running_tasks && static::AUTOMATIC_SCHEDULING) {
```

Comment:
Actually we need to know that there are no running or scheduled tasks so we need to fetch them. But there is another issue. The check has to be nested inside so that we end process same way even if automatic scheduling is disabled.